src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -186,13 +187,18 @@ protected override void ApplyResponseChallenge()
             if (Options.ProtocolValidator.RequireNonce)
             {
                 message.Nonce = Options.ProtocolValidator.GenerateNonce();
-                if (Options.NonceCache != null)
+                if (Options.CacheNonces)
                 {
-                    if (!Options.NonceCache.TryAddNonce(message.Nonce))
+                    if (await Options.NonceCache.GetAsync(message.Nonce) != null)
                     {
                         Logger.LogError(Resources.OIDCH_0033_TryAddNonceFailed, message.Nonce);
```

Comment:
Except that there are currently many designers that are not located in the `Properties` directory: `build resx` doesn't detect them and create new designers under `Properties`, which creates duplicate files. I could remove the existing designers, but that's not something I want to do as part of this PR.